Pass module arguments as argument instead of keeping it in runner

dhozac opened the pull request:
This cleans up self.runner.module_args by making it an argument to the functions that need it. In doing so, it also fixes running playbooks with -f 1, as they would previously run with whatever arguments the first node had.

It breaks the action_plugin API, but I think this is better to do now than later, and there's not really any good alternative. If backwards compatability is an issue, we could perhaps catch that exception and retry without it, but that seems ugly...

This will need to rebased on top of #1064 once that gets merged, but I figured I should get comments on the plugin API first.

dhozac committed:
Pass module arguments as argument instead of keeping it in runner

mpdehaan commented:
As action plugins are new for 0.8 I am not concerned with the API compat yet since it is not yet released.  Will look at this in a few days.  Thanks!

-- Michael

On Sep 21, 2012, at 4:56 AM, Daniel Hokka Zakrisson notifications@github.com wrote:

> This cleans up self.runner.module_args by making it an argument to the functions that need it. In doing so, it also fixes running playbooks with -f 1, as they would previously run with whatever arguments the first node had.
> 
> It breaks the action_plugin API, but I think this is better to do now than later, and there's not really any good alternative. If backwards compatability is an issue, we could perhaps catch that exception and retry without it, but that seems ugly...
> 
> This will need to rebased on top of #1064 once that gets merged, but I figured I should get comments on the plugin API first.
> 
> You can merge this Pull Request by running:
> 
>   git pull https://github.com/dhozac/ansible module_args-argument
> Or view, comment on, or merge it at:
> 
>   https://github.com/ansible/ansible/pull/1068
> 
> Commit Summary
> 
> Pass module arguments as argument instead of keeping it in runner
> File Changes
> 
> M lib/ansible/runner/**init**.py (45)
> M lib/ansible/runner/action_plugins/assemble.py (4)
> M lib/ansible/runner/action_plugins/async.py (3)
> M lib/ansible/runner/action_plugins/copy.py (8)
> M lib/ansible/runner/action_plugins/fetch.py (4)
> M lib/ansible/runner/action_plugins/normal.py (8)
> M lib/ansible/runner/action_plugins/raw.py (4)
> M lib/ansible/runner/action_plugins/template.py (8)
> Patch Links
> 
> https://github.com/ansible/ansible/pull/1068.patch
> https://github.com/ansible/ansible/pull/1068.diff
> —
> Reply to this email directly or view it on GitHub.

dhozac commented:
Okay, I have rebased this on top of devel, and I've also fixed some issues I found with daisychaining. As far as I can tell, this now works as intended.

mpdehaan commented:
Definitely much much cleaner too, merged, thanks!

mpdehaan commented:
(manually merged, pushed)
